src/sagas/account.js

```diff
@@ -27,31 +27,32 @@ import { getTokenName } from '../selectors/env';
 import { minDelayCall } from './helpers';
 
 export function * doSignup (action) {
-  const { formData } = action;
+  const { formData, nextRoute } = action;
+
   try {
     const response = yield minDelayCall(api.signup, formData);
 
     // Persist token for future page loads
     const tokenName = yield select(getTokenName);
-    new Cookies().set(tokenName, response.key, { path: '/' });
+    new Cookies().set(tokenName, response.key);
 
-    yield put({ type: SIGNUP_SUCCESS, ...response });
+    yield put({ type: SIGNUP_SUCCESS, response, nextRoute });
   } catch (error) {
     yield put({ type: SIGNUP_ERROR, error: error.response.data });
   }
 }
 
 export function* doLogin (action) {
-  const { formData } = action;
+  const { formData, nextRoute } = action;
 
   try {
     const response = yield minDelayCall(api.login, formData);
 
     // Persist token for future page loads
     const tokenName = yield select(getTokenName);
-    new Cookies().set(tokenName, response.key, { path: '/' });
+    new Cookies().set(tokenName, response.key);
 
-    yield put({ type: LOGIN_SUCCESS, ...response });
+    yield put({ type: LOGIN_SUCCESS, response, nextRoute });
```

Comment:
I believe this change breaks the account reducer since the response data is now under `response.<data>` (see the account reducer where it grabs the `user` and `key` values from the action). Have you tested this with the app?